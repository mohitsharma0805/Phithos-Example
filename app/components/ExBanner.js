import React from 'react';
import { View, Text, StyleSheet, Platform } from 'react-native';
import DeviceInfo from 'react-native-device-info';
import config from '../config';

const ExBanner = () => {
  const getDeviceType = () => {
    let deviceType = DeviceInfo.getModel();
    if (DeviceInfo.isEmulatorSync) {
      if (Platform.OS === 'ios') {
        deviceType = 'Simutator';
      } else {
        deviceType = 'Emulator';
      }
    }
    return deviceType;
  };

  return (
    <View style={styles.contentView}>
      <Text style={styles.text}>{getDeviceType()}</Text>
    </View>
  );
};

export default ExBanner;

const styles = StyleSheet.create({
  contentView: {
    justifyContent: 'center',
    height: 40,
    width: '100%',
    backgroundColor: config.colors.black,
  },
  text: { marginLeft: 20, color: config.colors.white, fontSize: 17 },
});
