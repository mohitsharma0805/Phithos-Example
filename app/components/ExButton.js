import React from 'react';
import { TouchableOpacity, Text, StyleSheet } from 'react-native';
import config from '../config';

const ExButton = ({ buttonStyle, text, onPress }) => {
  return (
    <TouchableOpacity onPress={onPress} style={[styles.button, buttonStyle]}>
      <Text style={styles.text}>{text}</Text>
    </TouchableOpacity>
  );
};

export default ExButton;

const styles = StyleSheet.create({
  button: {
    justifyContent: 'center',
    alignItems: 'center',
    marginHorizontal: 20,
    borderRadius: 5,
    height: 50,
    backgroundColor: config.colors.appTheme,
  },
  text: { color: config.colors.white, fontSize: 17 },
});
