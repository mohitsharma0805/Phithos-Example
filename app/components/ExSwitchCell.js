import React from 'react';
import { View, StyleSheet, Text, Switch } from 'react-native';
import config from '../config';

const ExSwitchCell = ({ switchValue, onSwitchValueChange }) => {
  const switchValueChanged = (value) => onSwitchValueChange?.(value);
  return (
    <View style={styles.container}>
      <View style={styles.wrapper}>
        <View style={styles.textWrapper}>
          <Text style={styles.title}>Banner Status</Text>
        </View>
        <Switch
          style={styles.switch}
          value={switchValue}
          onValueChange={switchValueChanged}
          trackColor={{
            true: config.colors.appTheme,
            false: null,
          }}
        />
      </View>
    </View>
  );
};

export default ExSwitchCell;

const styles = StyleSheet.create({
  container: { backgroundColor: config.colors.white },
  wrapper: { flexDirection: 'row', alignItems: 'center', height: 60 },
  textWrapper: { flex: 1 },
  title: {
    marginLeft: 16,
    paddingVertical: 2,
    fontSize: 17,
  },
  switch: { marginRight: 20, marginLeft: 16 },
});
