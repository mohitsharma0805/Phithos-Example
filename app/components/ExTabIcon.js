import React from 'react';
import { View, StyleSheet } from 'react-native';
import SimpleLineIcons from 'react-native-vector-icons/SimpleLineIcons';

const ExTabIcon = ({ containerStyle, iconName, iconStyle }) => {
  return (
    <View style={[styles.container, containerStyle]}>
      <SimpleLineIcons name={iconName} style={[styles.icon, iconStyle]} />
    </View>
  );
};

export default ExTabIcon;

const styles = StyleSheet.create({
  container: {
    width: 26,
    height: 26,
    margin: 5,
  },
  icon: { marginTop: 4, fontSize: 22 },
});
