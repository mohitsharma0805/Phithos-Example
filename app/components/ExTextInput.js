import React from 'react';
import { View, StyleSheet, TextInput } from 'react-native';
import config from '../config';

const ExTextInput = ({ viewStyle, ...props }) => {
  return (
    <View style={[styles.contentView, viewStyle]}>
      <TextInput {...props} style={styles.textInput} />
      <View style={styles.divider} />
    </View>
  );
};

export default ExTextInput;

const styles = StyleSheet.create({
  contentView: {
    marginLeft: 20,
    marginRight: 20,
    height: 50,
  },
  textInput: {
    flex: 1,
    marginTop: 12,
    height: 30,
    fontSize: 15,
    padding: 5,
  },
  divider: {
    marginTop: 5,
    height: 1,
    backgroundColor: config.colors.cellDivider,
  },
});
