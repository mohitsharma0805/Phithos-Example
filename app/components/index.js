export { default as ExButton } from './ExButton';
export { default as ExTextInput } from './ExTextInput';
export { default as ExBanner } from './ExBanner';
export { default as ExSwitchCell } from './ExSwitchCell';
