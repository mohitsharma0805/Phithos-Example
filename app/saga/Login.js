import { put, takeLatest } from 'redux-saga/effects';
import { User } from '../state';

export function* login(action) {
  // call api here
  yield put(User.saveUser(action.payload));
}

export function* watchLogin() {
  yield takeLatest('LOGIN', login);
}
