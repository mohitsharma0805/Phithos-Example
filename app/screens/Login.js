import React, { useState, useEffect } from 'react';
import { View, StyleSheet, SafeAreaView } from 'react-native';
import { useDispatch, useSelector } from 'react-redux';
import { StackActions } from '@react-navigation/native';
import { ExButton, ExTextInput, ExBanner } from '../components';
import config from '../config';
import { validateLoginDetails } from '../controls/Validator';
import { selectBannerStatus } from '../state/Settings';
import { selectUser } from '../state/User';

const Login = ({ navigation }) => {
  const dispatch = useDispatch();
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const bannerStatus = useSelector(selectBannerStatus);
  const user = useSelector(selectUser);

  useEffect(() => {
    loginSuccessfully();
  }, [user]);

  const loginSuccessfully = () => {
    if (Object.values(email).length > 0) {
      navigation.dispatch(StackActions.replace(config.routes.BOTTOM_TAB_NAVIGATION));
    }
  };

  const loginClicked = () => {
    if (validateLoginDetails(email, password)) {
      dispatch({ type: 'LOGIN', payload: { email, password } });
    }
  };

  return (
    <SafeAreaView style={styles.contentView}>
      {bannerStatus && <ExBanner />}
      <View style={styles.container}>
        <ExTextInput
          placeholder={config.strings.EMAIL}
          autoCapitalize={config.strings.NONE}
          returnKeyType={config.strings.NEXT}
          keyboardType={config.strings.EMAIL_ADDRESS_KEYBOARD_TYPE}
          blurOnSubmit={false}
          onChangeText={(text) => setEmail(text)}
        />
        <ExTextInput
          placeholder={config.strings.PASSWORD}
          secureTextEntry
          autoCapitalize={config.strings.NONE}
          returnKeyType={config.strings.DONE}
          keyboardType={config.strings.EMAIL_ADDRESS_KEYBOARD_TYPE}
          blurOnSubmit={false}
          onChangeText={(text) => setPassword(text)}
        />
        <ExButton buttonStyle={styles.button} text={config.strings.LOGIN} onPress={loginClicked} />
      </View>
    </SafeAreaView>
  );
};

export default Login;

const styles = StyleSheet.create({
  contentView: { flex: 1 },
  container: { marginTop: 125 },
  button: { marginTop: 50 },
});
