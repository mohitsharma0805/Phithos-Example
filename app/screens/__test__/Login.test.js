import React from 'react';
import { shallow } from 'enzyme';
import { SafeAreaView } from 'react-native';
import * as redux from 'react-redux';
import configureStore from 'redux-mock-store';
import createSagaMiddleware from 'redux-saga';
import Login from '../Login';
import { ExTextInput, ExButton } from '../../components';

describe('Songs', () => {
  describe('UI test cases', () => {
    const wrapper = shallow(<Login />);
    it('renders the main view', () => {
      expect(wrapper).toMatchSnapshot();
      expect(wrapper.find(SafeAreaView).length).toEqual(1);
      expect(wrapper.find(ExTextInput).length).toEqual(2);
      expect(wrapper.find(ExButton).length).toEqual(1);
    });
  });
});
