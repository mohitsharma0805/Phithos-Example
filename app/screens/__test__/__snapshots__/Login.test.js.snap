// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Songs UI test cases renders the main view 1`] = `ShallowWrapper {}`;
